executor/aggfuncs/aggfuncs.go

```diff
@@ -203,3 +203,16 @@ type SlidingWindowAggFunc interface {
 	// completely.
 	Slide(sctx sessionctx.Context, rows []chunk.Row, lastStart, lastEnd uint64, shiftStart, shiftEnd uint64, pr PartialResult) error
 }
+
+// MaxMinSlidingWindowAggFunc is the interface to evaluate the max/min agg function using sliding window
+type MaxMinSlidingWindowAggFunc interface {
+	// SetWindowStart sets the start position of window
+	SetWindowStart(start uint64)
+	// UpdatePartialResult is the same as AggFunc.UpdatePartialResult, providing such function to all MaxMinSlidingWindowAggFunc
+	UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (memDelta int64, err error)
+	// ResetPartialResult resets the partial result to the original state for a
+	// specific aggregate function. It converts the input PartialResult to the
+	// specific data structure which stores the partial result and then reset
+	// every field to the proper original state.
+	ResetPartialResult(pr PartialResult)
```

Comment:
I'm not sure that we really need `UpdatePartialResult` and `ResetPartialResult` here. It's repetition of the `AggFunc` interface.

Reply:
After rethinking, this two methods are indeed redundant. Removed.